src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpPossiblyDeclareAsNullableDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.PossiblyDeclareAsNullable
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpPossiblyDeclareAsNullableDiagnosticAnalyzer :
+        DiagnosticAnalyzer
+    {
+        private static readonly DiagnosticDescriptor s_possiblyDeclareAsNullableRule
+            = new DiagnosticDescriptor(
+            IDEDiagnosticIds.PossiblyDeclareAsNullableDiagnosticId,
+            title: new LocalizableResourceString(nameof(FeaturesResources.Variable_could_be_annotated_as_nullable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+            messageFormat: new LocalizableResourceString(nameof(FeaturesResources.Variable_0_could_be_annotated_as_nullable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+            DiagnosticCategory.CodeQuality,
+            DiagnosticSeverity.Info,
+            isEnabledByDefault: true,
+            description: new LocalizableResourceString(nameof(FeaturesResources.NullTestedVariablePossiblyDeclaredNullableDescription), FeaturesResources.ResourceManager, typeof(FeaturesResources)));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => ImmutableArray.Create(s_possiblyDeclareAsNullableRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.EqualsExpression, SyntaxKind.NotEqualsExpression,
+                SyntaxKind.IsPatternExpression, SyntaxKind.ConditionalAccessExpression);
+        }
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var fixableSymbol = IsFixable(context.Node, context.SemanticModel);
+            if (fixableSymbol != null)
+            {
+                context.ReportDiagnostic(Diagnostic.Create(s_possiblyDeclareAsNullableRule, context.Node.GetLocation(), fixableSymbol.Name));
```

Comment:
so, if i understand this correctly, all these variables will have `...` under them, but there is no way for the user to enable/disable this?  Is that the right experience we want here?